processor/fanoutconnector.go

```diff
@@ -19,12 +19,37 @@ import (
 
 	"github.com/open-telemetry/opentelemetry-collector/consumer"
 	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/internal/data"
 	"github.com/open-telemetry/opentelemetry-collector/oterr"
 )
 
 // This file contains implementations of Trace/Metrics connectors
 // that fan out the data to multiple other consumers.
 
+// CreateMetricsFanOutConnector creates a connector based on provided type of trace consumer.
+// If any of the wrapped metrics consumers are of the new type, use metricsFanOutConnectorV2,
+// otherwise use the old type connector.
+func CreateMetricsFanOutConnector(mcs []consumer.BaseMetricsConsumer) consumer.BaseMetricsConsumer {
+	metricsConsumers := make([]consumer.MetricsConsumer, 0, len(mcs))
+	metricsConsumersV2 := make([]consumer.MetricsConsumerV2, 0, len(mcs))
+	anyMetricsConsumersV2 := false
+	for _, mc := range mcs {
+		metricsConsumer := mc.(consumer.MetricsConsumer)
```

Comment:
Shouldn't this be?
```suggestion
		metricsConsumer, ok := mc.(consumer.MetricsConsumer)
		if ok ...
```
Otherwise if `mc` is `consumer.MetricsConsumerV2` this will panic.

Reply:
Correct. Caught it when adding tests as well. Moved it under the right condition.